app/src/main/java/com/example/carhive/Presentation/initial/Register/view/ThirdRegisterFragment.kt

```diff
@@ -37,40 +42,59 @@ class ThirdRegisterFragment : Fragment() {
-            if (uri != null) {
-                imageUri = uri
-                binding.selectedImageView.visibility = View.VISIBLE
-                binding.selectedImageView.load(uri) // Carga la imagen seleccionada
-                binding.selectedImageView.load(uri) {
-                    transformations(coil.transform.CircleCropTransformation()) // Aplica recorte circular
+        // Initialize the camera launcher with intent for front camera
+        cameraLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == android.app.Activity.RESULT_OK) {
+                imageUri?.let { uri ->
+                    binding.selectedImageView.visibility = View.VISIBLE
+                    binding.selectedImageView.load(uri) {
+                        transformations(coil.transform.CircleCropTransformation())
+                    }
                 }
+                binding.finishRegistrationButton.isEnabled = true
             }
-            // Habilitar el botón si hay una imagen seleccionada
-            binding.finishRegistrationButton.isEnabled = uri != null
         }
 
         binding.chooseImageButton.setOnClickListener {
-            launcher.launch("image/*") // Lanza el selector de imágenes
+            imageUri = createImageUri(requireContext())
+            imageUri?.let { uri ->
+                val cameraIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE).apply {
+                    putExtra(MediaStore.EXTRA_OUTPUT, uri)
+                    // Request front camera
+                    putExtra("android.intent.extras.CAMERA_FACING", 1) // For older devices
+                    putExtra("android.intent.extras.LENS_FACING_FRONT", 1) // For newer devices
+                    putExtra("android.intent.extra.USE_FRONT_CAMERA", true) // Extra fallback
+                }
+                cameraLauncher.launch(cameraIntent)
+            }
         }
 
         binding.finishRegistrationButton.setOnClickListener {
             imageUri?.let { uri ->
-                viewModel.uploadProfileImage(uri) // Llama al método de carga de la imagen
-                findNavController().navigate(R.id.action_thirdRegisterFragment_to_fortRegisterFragment) // Cambia a tu siguiente fragmento
+                viewModel.uploadProfileImage(uri)
+                findNavController().navigate(R.id.action_thirdRegisterFragment_to_fortRegisterFragment)
             }
         }
 
         binding.backToSecondPartLink.setOnClickListener {
-            findNavController().navigate(R.id.action_thirdRegisterFragment_to_secondRegisterFragment) // Cambia a tu fragmento anterior
+            findNavController().navigate(R.id.action_thirdRegisterFragment_to_secondRegisterFragment)
         }
     }
 
+    private fun createImageUri(context: Context): Uri? {
+        val image = File(context.filesDir, "camera_photo.jpg")
```

Comment:
why  the image save how .jpg ? , Because don´t  her name is the Id of the user?

Reply:
Because you would save that important information such as the name of the photo, just so you know, that photo is saved in the external storage of the device, it is not that it is saved in the database, if we did what you say we would have a great lack of security,

Reply:
For the next one, I would appreciate if you could ask me about the operation so that I can see if I am really wrong or if it is necessary what you are telling me in order to avoid misunderstandings and to reject pull requests without grounds. ;)

Reply:
You tell me that it is saved locally on the device, but then if it is only saved on the user's device with which the photo was taken, then how are other users going to be able to see the user's photo in the user's profile? 
If you are telling me that it is saved locally

Reply:
This photo is also uploaded to the database, but it has nothing to do with that part, this is only for the operation of the photo in real time, the functionality of uploading the photo to the database was already done, no changes were made in that logic because it is not necessary, for this reason it does not matter how this photo is called.

Reply:
ok I don't feel like fighting